lang_qc/db/qc_connection.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2022 Genome Research Ltd.
+#
+# Author: Adam Blanchet <ab59@sanger.ac.uk>
+#
+# This file is part of npg_langqc.
+#
+# npg_langqc is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, Session
+
+config = {"QCDB_URL": os.environ.get("QCDB_URL"), "TEST": os.environ.get("LANGQC_MODE")}
+
+if config["TEST"]:
+    config["QCDB_URL"] = "sqlite+pysqlite:///:memory:"
```

Comment:
If we're demanding a MySQL instance for the warehouse tests, is it wise to bring another DB engine in?

Reply:
Probably not, and the conversation above made me realise that the `sqlite` driver isn't specified as required. So `pysqlite` doesn't get installed anyway.
The `PyMySQL` one is only installed because it's a dependency of `ml-warehouse`. That's not great in the unlikely case we stop using `ml-warehouse`

Reply:
Since LANGQC_MODE is no longer a thing after #20 gets in, this PR should probably happen after it. That way we can set the db getter override in the TestClient (and probably get it to use the same DB server).